core/datastore/src/test/kotlin/com/google/samples/apps/nowinandroid/core/datastore/NiaPreferencesDataSourceTest.kt

```diff
@@ -32,11 +37,34 @@ class NiaPreferencesDataSourceTest {
 
     private lateinit var subject: NiaPreferencesDataSource
 
+    // A DataStore implementation that throws an IOException when accessed
+    private val ioExceptionThrowingDataStore = object : DataStore<UserPreferences> {
+        override val data: Flow<UserPreferences>
+            get() = flow { throw IOException("Failed to read proto") }
+
+        override suspend fun updateData(transform: suspend (t: UserPreferences) -> UserPreferences): UserPreferences {
+            throw Exception("Not needed for this test")
+        }
+    }
+
     @Before
     fun setup() {
         subject = NiaPreferencesDataSource(InMemoryDataStore(UserPreferences.getDefaultInstance()))
     }
 
+    @Test
+    fun userData_emitDefault_whenDataStoreThrowsIOException() =
+        testScope.runTest {
+            // Given: NiaPreferencesDataSource with ioException throwing datastore
+            val dataSource = NiaPreferencesDataSource(ioExceptionThrowingDataStore)
+
+            // When: Retrieving user data from the data source
+            val actualUserData = dataSource.userData.first()
+
+            // Then: The default user data is returned
+            assertEquals(subject.userData.first(), actualUserData)
```

Comment:
nit: self-explanatory comment are not useful. We already have the test method name that explain what the test does. Comments are important for explaining the why, not the what. :)